### PR TITLE
Kusama Extrinsic signing

### PR DESCRIPTION
**Pull request by hewigovens**
- [x] Mortal era
- [x] Bond/Unbond/BondExtra/WithdrawUnbonded tests
- [x] very long vector

### CODE REVIEW

File: src/Polkadot/ScaleCodec.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+inline Data encodeCompact(CompactInteger value) {
+    auto data = Data{};
+
+    if (value < kMinUint16) {
+        auto v = value.convert_to<uint8_t>() << 2u;
+        data.push_back(static_cast<uint8_t>(v));
+        return data;
+    } else if (value < kMinUint32) {
+        auto v = (value.convert_to<uint16_t>() << 2u);
+        v += 0b01; // set 0b01 flag
```
Comment (catenocrypt) [inline src/Polkadot/ScaleCodec.h]: Since other values are hex, I would use 0x01, which is the same, but more readable here.

Comment (hewigovens) [reply src/Polkadot/ScaleCodec.h]: done

File: src/proto/Polkadot.proto [hunk 2]
```diff
@@ -3,24 +3,80 @@ syntax = "proto3";
...
+
+enum RewardDestination {
+    STAKED = 0;
+    STASH = 1;
+    CONTROLLER = 2;
+}
+
+message Era {
+    uint64 period = 1;
+    uint64 phase = 2;
```
Comment (vikmeup) [inline src/proto/Polkadot.proto]: what is `Era`?

File: src/proto/Polkadot.proto [hunk 3]
```diff
@@ -3,24 +3,80 @@ syntax = "proto3";
 package TW.Polkadot.Proto;
 option java_package = "wallet.core.jni.proto";
 
-message ExtrinsicsTransfer {
-    string to_address = 1;
-    bytes value = 2;
+enum Network {
+    POLKADOT = 0;
```
Comment (vikmeup) [inline src/proto/Polkadot.proto]: do we need to define it here or use fetch from the network?

Comment (hewigovens) [reply src/proto/Polkadot.proto]: I think we can fetch it from rpc, but we need this to parse correct address, we only support two ss58 types here

File: src/proto/Polkadot.proto (hunk 2 above)
Comment (hewigovens) [reply src/proto/Polkadot.proto]: https://wiki.polkadot.network/docs/en/glossary#era

**Pull request merged**